core/src/main/java/bisq/core/dao/state/DaoStateService.java

```diff
@@ -743,29 +743,75 @@ public long getTotalAmountOfUnLockedTxOutputs() {
+                Optional<SpentInfo> optionalSpentInfo = getSpentInfo(lockupTxOutput);
+                checkArgument(optionalSpentInfo.isPresent(), "optionalSpentInfo must be present");
+                String unlockTxId = optionalSpentInfo.get().getTxId();
+                if (isUnlockingAndUnspent(unlockTxId)) {
+                    // We found the unlock tx is still not spend
+                    log.warn("lockupTxOutput {} is still unspent. We confiscate it.", lockupTxOutput.getKey());
+                    getUnspentUnlockingTxOutputsStream()
+                            .filter(txOutput -> txOutput.getTxId().equals(unlockTxId))
+                            .forEach(this::confiscateBond);
+                } else {
+                    // We could be more radical here and confiscate the output if it is unspent but lock time is over,
+                    // but its probably better to stick to the rules that confiscation can only happen before lock time
+                    // is over.
+                    log.warn("We could not confiscate the bond because the unlock tx was already spent or lock time " +
+                            "has exceeded. unlockTxId={}", unlockTxId);
+                }
+            }
+        } else {
+            log.warn("No lockupTxOutput found for lockupTxId {}", lockupTxId);
         }
-        getTxOutputStream()
-                .filter(txOutput -> isUnspent(txOutput.getKey()))
-                .filter(txOutput -> txOutput.getTxOutputType() == TxOutputType.LOCKUP_OUTPUT ||
-                        (isUnlockTxOutputAndLockTimeNotOver(txOutput)))
-                .filter(txOutput -> {
-                    Optional<byte[]> hash = getLockupHash(txOutput);
-                    return hash.isPresent() && Arrays.equals(hash.get(), confiscateBond.getHash());
-                })
-                .forEach(this::applyConfiscateBond);
     }
 
-    public void applyConfiscateBond(TxOutput txOutput) {
+    private void confiscateBond(TxOutput txOutput) {
+        // Can be txOutput from lockup or unlock tx
+        log.warn("txOutput {} added to confiscatedTxOutputMap.", txOutput.getKey());
         daoState.getConfiscatedTxOutputMap().put(txOutput.getKey(), txOutput);
+    }
+
+    public boolean isConfiscated(TxOutputKey txOutputKey) {
+        return daoState.getConfiscatedTxOutputMap().containsKey(txOutputKey);
+    }
 
-        // TODO SQ TxOutputType is immutable after parsing
-        // We need to add new checks if a txo is not confiscated by using the map similar like utxo map
-        // txOutput.setTxOutputType(TxOutputType.BTC_OUTPUT);
+    public boolean isConfiscatedLockupTxOutput(String lockupTxId) {
+        return getLockupTxOutput(lockupTxId)
+                .map(lockupTxIdxOutput -> isConfiscated(lockupTxIdxOutput.getKey()))
+                .orElse(false);
     }
 
+    public boolean isConfiscatedUnlockTxOutput(String unlockTxId) {
```

Comment:
What is the motivation to do confiscation on both lockupTxOutput and unlockTxOutput? Wouldn't it be clearer to only use the lockupTxOutput for confiscation?

Reply:
If the user has created already the unlock tx confiscating the lockup txo will not have any effect. For identifying we use the lockup tx.
But after a second thought, you might be right. If the lockup txo is confiscated all follow up are invalid anyway. Need to check again and see if we are filtering by unspent state...
Would need more testing anyway...

Reply:
That's what I was thinking, no point in targeting the individual unlock tx since the bsq is already confiscated in the lockup. But perhaps it's needed to mark the unlock txouput as confiscated at the point of parsing the vote result if it's already being unlocked.